### PR TITLE
move go-ws-transport here

### PR DESCRIPTION
**Pull request by marten-seemann**
Part of #1187.

### CODE REVIEW

File: p2p/transport/websocket/websocket_test.go [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+	"github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-core/sec"
+	"github.com/libp2p/go-libp2p-core/sec/insecure"
+	"github.com/libp2p/go-libp2p-core/test"
+	"github.com/libp2p/go-libp2p-core/transport"
+
+	csms "github.com/libp2p/go-conn-security-multistream"
+	mplex "github.com/libp2p/go-libp2p-mplex"
```
Comment (vyzo) [inline p2p/transport/websocket/websocket_test.go:27]: we should probably wait for these to be moved first.

Comment (marten-seemann) [reply p2p/transport/websocket/websocket_test.go:27]: Why? go-libp2p depending on external repos is ok (the other way around would not).

Comment (vyzo) [reply p2p/transport/websocket/websocket_test.go:27]: sure, but we'll have to remember to fix it!

Comment (marten-seemann) [reply p2p/transport/websocket/websocket_test.go:27]: Of course. I do a `sed` for the import path every time I move a repo.

Comment (vyzo) [reply p2p/transport/websocket/websocket_test.go:27]: alright, fair enough.

**Review by vyzo — APPROVED** [review]

**Pull request merged**